pkg/util/apigroup.go

```diff
@@ -0,0 +1,122 @@
+package util
+
+import (
+	"regexp"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	clusterv1alpha1 "github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1"
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+	workv1alpha1 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha1"
+)
+
+// ResourceConfig represents which Group GroupVersion GroupVersionKind is disabled.
+type ResourceConfig struct {
+	Group            map[string]bool
+	GroupVersion     map[schema.GroupVersion]bool
+	GroupVersionKind map[schema.GroupVersionKind]bool
+}
+
+// NewResourceConfig to create ResourceConfig
+func NewResourceConfig() *ResourceConfig {
+	r := &ResourceConfig{Group: map[string]bool{},
+		GroupVersion:     map[schema.GroupVersion]bool{},
+		GroupVersionKind: map[schema.GroupVersionKind]bool{}}
+	// disable karmada group by default
+	r.DisableGroup(clusterv1alpha1.GroupVersion.Group)
+	r.DisableGroup(policyv1alpha1.GroupVersion.Group)
+	r.DisableGroup(workv1alpha1.GroupVersion.Group)
+	return r
+}
+
+// Parse to parse tokens to ResourceConfig
+func (r *ResourceConfig) Parse(c string) {
+	// v1/Node,Pod;networking.k8s.io/v1beta1/Ingress,IngressClass
+	// group networking.k8s.io
+	// group version networking.k8s.io/v1
+	// group version kind networking.k8s.io/v1/Ingress
+	// corev1 has no group
+	tokens := strings.Split(c, ";")
+	for _, token := range tokens {
+		r.parseSingle(token)
+	}
+}
+
+func (r *ResourceConfig) parseSingle(token string) {
+	var groupVersionKindPattern = regexp.MustCompile(`^([\w.]+)/(\w+)/([\w,]*)`)
```

Comment:
Not quite sure how `[.]` is parsed in the golang regexp, do we need escape with `\`? I mean `[\w\.]`

Reply:
Regexp is a little bit complicated and hard to understand.
Can we parse it like the [ParseGroupVersion](https://github.com/kubernetes/kubernetes/blob/696d0f5772d030a37ea67056b23e75330321a345/staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version.go#L211)?

Reply:
We need a unit test here by the way.

Reply:
I will dig into it.Regex is complicated indeed.